src/ui/popup/disallowed.tsx

```diff
@@ -0,0 +1,69 @@
+import ClonedSong from '@/core/object/cloned-song';
+import { ManagerTab } from '@/core/storage/wrapper';
+import { Resource, Show, createMemo, createSignal } from 'solid-js';
+import styles from './popup.module.scss';
+import optionComponentStyles from '../options/components/components.module.scss';
+import { t } from '@/util/i18n';
+import MusicOff from '@suid/icons-material/MusicOffOutlined';
+import Send from '@suid/icons-material/Send';
+import { sendBackgroundMessage } from '@/util/communication';
+
+export default function Disallowed(props: { tab: Resource<ManagerTab> }) {
+	const [isLoading, setLoading] = createSignal(false);
+	const song = createMemo(() => {
+		const rawTab = props.tab();
+		if (!rawTab) return null;
+		const rawSong = rawTab.song;
+		if (!rawSong) return null;
+		return new ClonedSong(rawSong, rawTab.tabId);
+	});
+
+	return (
+		<Show when={!isLoading()}>
+			<div class={styles.widePopup}>
+				<MusicOff class={styles.bigIcon} />
+				<h1>{t('disallowedHeader')}</h1>
+				<p>
+					{t('disallowedDesc1', [
+						song()?.getArtist() ?? '???',
+						song()?.getTrack() ?? '???',
+					])}
+				</p>
+				<Show
+					when={song()?.metadata.label === 'YouTube'}
+					fallback={
```

Comment:
Yes, show/fallback is the idiomatic if/else in solid.
Solid has a switch/match that is what you expect, however I don’t expect we will ever add anything more than youtube here.
Keep in mind that every single connector we add a separate copy for here will need to have their messages translated, so I don’t think we should add special text for connectors unless there is a huge reason to do so (which there is for youtube because we have youtube specific settings that often trigger this).

Reply:
Here is solidjs interactive tutorial on control flow (theres a few of them, you can see in the table of contents dropdown) https://www.solidjs.com/tutorial/flow_show
Should just take a couple minutes to go through and get a basic understanding of the control flow mechanisms, especially if you’ve already used some other framework that uses jsx, such as react.